clippy_lints/src/format_args.rs

```diff
@@ -86,11 +149,64 @@ impl<'tcx> LateLintPass<'tcx> for FormatArgs {
                     check_format_in_format_args(cx, outermost_expn_data.call_site, name, arg.param.value);
                     check_to_string_in_format_args(cx, name, arg.param.value);
                 }
+                if do_inline && let Some(inline_spans) = inline_spans  {
+                    span_lint_and_then(
+                        cx,
+                        NEEDLESS_FORMAT_ARGS,
+                        outermost_expn_data.call_site,
+                        "variables can be used directly in the `format!` string",
+                        |diag| {
+                            diag.multipart_suggestion("change this to", inline_spans, Applicability::MachineApplicable);
+                        },
+                    );
+                }
             }
         }
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+enum ParamType {
+    Argument,
+    Width,
+    Precision,
+}
```

Comment:
This is pretty useful, I think it'd be worth adding to `FormatParam` (maybe with a different name so there aren't both `kind` and `type` fields), that way you could have like `for param in format_args.params() { check_inline(..) }`

Reply:
thx, yeah, makes things cleaner